File: Data_Structure_through_C/00_Sorting_00/Bubble_sort.c
#include <stdio.h>
#define MAX 100

int main(void)
{
	int arr[MAX], i, j, n, temp, xchanges;
	printf("Enter the number of elements : ");
	scanf("%d", &n);
	for (i = 0; i < n; i++)
	{
		printf("Enter element %d :  ", i + 1);
		scanf("%d", &arr[i]);
	}
	//bubble sort
	for (i = 0; i < n - 1; i++)
	{
		xchanges = 0;
		for (j = 0; j < n - i - 1; j++)
		{
			if (arr[j] > arr[j + 1])
			{
				temp = arr[j];
				arr[j] = arr[j + 1];
				arr[j + 1] = temp;
				xchanges++;
			}
		}
		if (xchanges == 0)
			break;
	}
	printf("Sorted list is : ");
	for (i = 0; i < n; i++)
		printf("%d  ", arr[i]);
	printf("\n");
	return 0;
}

File: Algorithms_py/Bubble_sort.py
print("This program sorts the given array using bubble sort technique")
n=int(input("Enter the number of elements you want to insert in the array "))
DATA=[]
for i in range(0,n):
    m=int(input("Enter element "))
    DATA.append(m)

print("The inserted array is", DATA)

for j in range(0,n):
    for k in range(0,n-j-1):
        if DATA[k]>DATA[k+1]:
            t=DATA[k]
            DATA[k]=DATA[k+1]
            DATA[k+1]=t

print("Sorted array", DATA)
